### PR TITLE
derive all the error traits instead of using hand-rolled trait implementations

### PR DESCRIPTION
**Pull request by BrianHicks**

### CODE REVIEW

**Comment by BrianHicks** [comment]
note: kinda-sorted replaced by https://github.com/NoRedInk/firstaide/pull/8

**Comment by BrianHicks** [comment]
well actually it totally replaces it, huh? But this was a necessary step in reducing code size and it's part of the history so I guess it's gotta stay!

**Comment by BrianHicks** [comment]
@caiquefigueiredo wanna pair-review this sometime today?

**Comment by caiquefigueiredo** [comment]
@BrianHicks I'd love to, yes

File: src/config.rs [hunk 1]
```diff
@@ -1,51 +1,34 @@
...
-    }
-}
+    #[error("configuration file not valid: {0}")]
+    Invalid(#[from] toml::de::Error),
 
-impl From<toml::de::Error> for Error {
-    fn from(error: toml::de::Error) -> Self {
-        Error::Invalid(error)
-    }
+    #[error("couuld not use configuration: {0}")]
```
Comment (caiquefigueiredo) [inline src/config.rs]:
```suggestion
    #[error("could not use configuration: {0}")]
```

**Review by caiquefigueiredo — APPROVED** [review]

**Pull request merged**